src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/H2Engine.java

```diff
@@ -385,10 +385,14 @@ protected MappedEntity createPreparedStatementForInserts(final DbEntity entity)
         final List<String> values = new ArrayList<>();
         final List<String> columnsWithAutoInc = new ArrayList<>();
         final List<String> valuesWithAutoInc = new ArrayList<>();
+        String columnWithAutoIncName = null;
         for (final DbColumn column : entity.getColumns()) {
             columnsWithAutoInc.add(quotize(column.getName()));
```

Comment:
Just noticed now... but why do we add all columns to this list, even the ones that aren't auto_increment?

Reply:
I know its weird, but not adding all the columns to that list, make some tests fail.

Reply:
Have you analyzed the test so see if they need to be adapted is this expected?

Reply:
That's what I'm trying to understand right now.

Reply:
Yes it's strange reading this code and watching it adding all the columns to a list named columnWithAutoIncName... I don't know this code but please check if this is ok. Maybe there is some place where it's being done a difference between the normal columns and columnsWithAutoInc in order to create a correct prepared statement :man_shrugging:

Reply:
Well I tried to figure that out, but it seems that when we only add the columns that contain auto incremented values in those lists `columnsWithAutoInc` and `valuesWithAutoInc`, the `MappedEntity` used when binding it to the prepared statement, it always needs to contain all the columns that an `EntityEntry` might have.

Reply:
Otherwise, it can't bind the properly and a `SQLException` is thrown.